pkg/registry/service/rest.go

```diff
@@ -157,7 +157,7 @@ func (rs *REST) Create(ctx api.Context, obj runtime.Object) (runtime.Object, err
 		}
 	}
 
-	if featuregate.DefaultFeatureGate.ExternalTrafficLocalOnly() && shouldCheckOrAssignHealthCheckNodePort(service) {
```

Comment:
The feature gate check already exists in that function being called (`shouldCheckOrAssignHealthCheckNodePort), so removed the redundant check here.

https://github.com/girishkalele/kubernetes/blob/master/pkg/registry/service/rest.go#L458